fix: Push status - If there are remote commits, force push is required

estib-vega opened the pull request:
If there are upstream commits in a branch, despite it having local or local-and-remote commits, pushing would remove the remote only commits and, in turn, require a force push

estib-vega committed:
fix: Push status - If there are remote commits, force push is required

If there are upstream commits in a branch, despite it having local or local-and-remote commits, pushing would remove the remote only commits and, in turn, require a force push

Byron commented:
If you have a specific case in mind, a test would be needed for that as well to pin it down. Even if existing tests break it's often best to add another one with the specific scenario you are fixing. It might overlap, but it's likely subtly different so it's definitely not a waste.

estib-vega committed:
test: adapt push status for upstream commits

Update test assertions so that branches with upstream (remote-only) commits now expect push_status: UnpushedCommitsRequiringForce.

Previously, these were expected as NothingToPush. Test code is updated for integrate_with_merges.rs, legacy.rs, and mod.rs accordingly.

Byron commented on crates/but-workspace/tests/workspace/ref_info/with_workspace_commit/journey/integrate_with_merges.rs:
```diff
@@ -492,7 +492,7 @@ fn remote_ahead() -> anyhow::Result<()> {
                             commits_outside: None,
                             metadata: "None",
-                            push_status: NothingToPush,
+                            push_status: UnpushedCommitsRequiringForce,
```
Indeed! I was looking at it an still thought there should be nothing to push, but it's true that pushing would require force here so the test was clearly wrong.
I have a feeling that in future, there will be more inter-op between "integrate remote" and these push buttons - after all one would want to make a fast-forward here really easy to do, while still allowing a force-push behind a safety check.
